extensions/wikia/EmergencyBroadcastSystem/EmergencyBroadcastSystemController.class.php

```diff
@@ -64,4 +65,68 @@ protected function hasNonPortableInfoBoxes() {
 		// TODO: Actually check for non portable infoboxes
 		return true;
 	}
+
+	protected function buildSurveyUrl() {
+		global $wgContLang;
+
+		$context = $this->getContext();
+		$userName = $context->getUser()->getName();
+		$userLanguage = $context->getLanguage();
+
+		$lang = $userLanguage->mCode;
+		$supportedLanguages = ['en', 'de', 'es', 'fr', 'it', 'ja', 'pl', 'pt-br', 'ru', 'zh'];
+
+		if ( !in_array( $lang, $supportedLanguages ) ) {
+			$lang = $wgContLang->mCode;
+		}
+
+		$url = $this->googleFormUrl( $lang, $userName);
+
+		return $url;
+	}
+
+	protected function googleFormUrl( $language, $userName ) {
+		global $wgServer;
+
+		preg_match( '/\/\/(.*)/', $wgServer, $matches );
```

Comment:
There "must" be some better way to figure out wikia domain - can you look for it. The less regex usually the better.